src/crimson/osd/main.cc

```diff
@@ -47,7 +47,7 @@ int main(int argc, char* argv[])
 
   args.insert(begin(args), argv[0]);
   try {
-    return app.run_deprecated(args.size(), const_cast<char**>(args.data()), [&] {
+    return app.run(args.size(), const_cast<char**>(args.data()), [&] {
```

Comment:
the application doesn't exit when the callback returns,  it has to be explicitly stopped by calling engine_exit().  So here we use run, it will call engine_exit() automatically.